crates/ev-precompiles/src/mint.rs

```diff
@@ -0,0 +1,343 @@
+// Mint precompile
+
+use alloy::{sol, sol_types::SolInterface};
+use alloy_evm::{
+    precompiles::{Precompile, PrecompileInput},
+    revm::precompile::{PrecompileError, PrecompileId, PrecompileResult},
+    EvmInternals, EvmInternalsError,
+};
+use alloy_primitives::{address, Address, Bytes, U256};
+use revm::precompile::PrecompileOutput;
+use std::sync::OnceLock;
+
+sol! {
+    interface INativeToken {
```

Comment:
where does this solidity code get produced?

Would need to import this interface from the token router

Reply:
I just added this as an interface dep. I guess this is enough

```solidity
// SPDX-License-Identifier: Apache-2.0
pragma solidity ^0.8.24;

/// @notice Minimal interface for the NativeToken precompile exposed by the host chain.
interface INativeToken {
    /// @notice Mints `amount` of native tokens into `account`.
    function mint(address account, uint256 amount) external;

    /// @notice Burns `amount` of native tokens from `account`.
    function burn(address account, uint256 amount) external;
}

```